lib/private/Repair/NC29/ValidateAccountProperties.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OC\Repair\NC29;
+
+use InvalidArgumentException;
+use OCP\Accounts\IAccountManager;
+use OCP\IUser;
+use OCP\IUserManager;
+use OCP\Migration\IOutput;
+use OCP\Migration\IRepairStep;
+use Psr\Log\LoggerInterface;
+
+class ValidateAccountProperties implements IRepairStep {
+
+	private const PROPERTIES_TO_CHECK = [
+		IAccountManager::PROPERTY_PHONE,
+		IAccountManager::PROPERTY_WEBSITE,
+		IAccountManager::PROPERTY_TWITTER,
+		IAccountManager::PROPERTY_FEDIVERSE,
+	];
+
+	public function __construct(
+		private IUserManager $userManager,
+		private IAccountManager $accountManager,
+		private LoggerInterface $logger,
+	) {
+	}
+
+	public function getName(): string {
+		return 'Validate account properties and store phone numbers in a known format for search';
+	}
+
+	public function run(IOutput $output): void {
+		$numRemoved = 0;
+
+		$this->userManager->callForSeenUsers(function (IUser $user) use (&$numRemoved) {
+			$account = $this->accountManager->getAccount($user);
+			$properties = array_keys($account->jsonSerialize());
+
+			// Check if there are some properties we can sanitize - reduces number of db queries
+			if (empty(array_intersect($properties, self::PROPERTIES_TO_CHECK))) {
+				return;
+			}
+
+			// Limit the loop to the properties we check to ensure there are no infinite loops
+			$iteration = count(self::PROPERTIES_TO_CHECK);
```

Comment:
```suggestion
			$iteration = count(self::PROPERTIES_TO_CHECK) + 1;
```
I think one more is needed, my rationale here being that if we check only 1 property, we may need 2 loop iterations, one to check, and a second one to check with the emptied value.